src/mpid/ch4/netmod/ofi/ofi_coll_select_container_parsers.c

```diff
@@ -0,0 +1,108 @@
+
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2006 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2011-2016 Intel Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
+ */
+
+#include <mpidimpl.h>
+#include "ofi_noinline.h"
+
+int MPIDU_SELECTION_MAX_OFI_ALGORITHM_INDICES[MPIDU_SELECTION_COLLECTIVES_MAX]
+    = {
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+};
+
+const char **MPIDU_SELECTION_OFI_ALGORITHMS[] = {
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+};
+
+void MPIDI_OFI_algorithm_parser(MPIDU_SELECTION_coll_id_t coll_id, int *cnt_num,
```

Comment:
Do we need to define a parser in every part of the device? I see, for instance, that UCX netmod contains an algorithm parser declaration, but there is no implementation like there is with OFI and POSIX?

Reply:
The parser function at CH4, OFI and POSIX are similar. But they iterate on the algorithms array from each layer and I had to declare it separately in all layers. I didn't add selection in UCX and just declared the function to fix compile error. I'm not thrilled about this part of the code as well. We can talk offline to see if we can refactor this.